lib/components/logIn.dart

```diff
@@ -33,8 +44,36 @@ class _LogInScreenState extends State<LogInScreen> {
     super.dispose();
   }
 
+  Future uploadProfilePic(BuildContext context) async {
+    // Profile Image name Same as user id
+    String fileName = FirebaseAuth.instance.currentUser.uid;
+    Reference firebaseStorageRef =
+        FirebaseStorage.instance.ref().child(fileName);
+    UploadTask uploadTask = firebaseStorageRef.putFile(_image);
+
+    ScaffoldMessenger.of(context).showSnackBar(SnackBar(
+        duration: Duration(seconds: 5),
+        content: Text('Wait...Picture is uploading')));
+    await uploadTask.whenComplete(() {
+      print("Picture Uploaded");
+      ScaffoldMessenger.of(context)
+          .showSnackBar(SnackBar(content: Text('Picture Uploaded')));
+    });
+  }
+
+  Future<File> urlToFile(String imageUrl) async {
+    var rng = new Random();
+    Directory tempDir = await getTemporaryDirectory();
+    String tempPath = tempDir.path;
+    File file = new File('$tempPath' + (rng.nextInt(100)).toString() + '.png');
+    http.Response response = await http.get(imageUrl);
+    await file.writeAsBytes(response.bodyBytes);
+    return file;
+  }
+
   @override
   Widget build(BuildContext context) {
+    var height = MediaQuery.of(context).size.height;
```

Comment:
Try to use `final` for immutable data.

`final height = MediaQuery.of(context).size.height`;